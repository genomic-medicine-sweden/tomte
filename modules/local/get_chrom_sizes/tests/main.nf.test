nextflow_process {

    name "Test Process GET_CHROM_SIZES"
    script "modules/local/get_chrom_sizes/main.nf"
    process "GET_CHROM_SIZES"
    tag "modules_local"
    tag "modules"
    tag "get_chrom_sizes"

    test("Should run without failures") {

        when {
            params {
                modules_testdata_base_path = 'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/'
            }
            process {
                """
                input[0] = [
                    [ id:'reference' ], // meta map
                    file(params.modules_testdata_base_path +'genomics/sarscov2/genome/genome.fasta.fai', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
