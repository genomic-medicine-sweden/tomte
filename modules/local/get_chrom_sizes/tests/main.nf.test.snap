{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "genome.fasta.fai.sizes:md5,a57c401f27ae5133823fb09fb21c8a3c"
                ],
                "1": [
                    "versions.yml:md5,5e54f71eb8c46f4afb44c4626e8a4c0a"
                ],
                "sizes": [
                    "genome.fasta.fai.sizes:md5,a57c401f27ae5133823fb09fb21c8a3c"
                ],
                "versions": [
                    "versions.yml:md5,5e54f71eb8c46f4afb44c4626e8a4c0a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-15T16:22:07.505881"
    }
}